src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs

```diff
@@ -931,8 +931,8 @@ public struct MINMAXINFO
         [DllImport("user32.dll", EntryPoint = "MapVirtualKeyW")]
         public static extern uint MapVirtualKey(uint uCode, uint uMapType);
 
-        [DllImport("user32.dll", EntryPoint = "GetMessageW")]
-        public static extern sbyte GetMessage(out MSG lpMsg, IntPtr hWnd, uint wMsgFilterMin, uint wMsgFilterMax);
+        [DllImport("user32.dll", EntryPoint = "GetMessageW", SetLastError = true)]
```

Comment:
I wonder, do we have any performance degradation with "SetLastError = true"? GetMessage is called quite often.

Reply:
This is only needed for a debug build

Reply:
this is banchmark on my machine
BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19041.630 (2004/?/20H1)
Intel Core i7-7700K CPU 4.20GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.1.401
  [Host]     : .NET Core 3.1.9 (CoreCLR 4.700.20.47201, CoreFX 4.700.20.47203), X86 RyuJIT  [AttachedDebugger]
  DefaultJob : .NET Core 3.1.9 (CoreCLR 4.700.20.47201, CoreFX 4.700.20.47203), X86 RyuJIT


|           Method |     Mean |    Error |   StdDev |   Median | Ratio | RatioSD |
|----------------- |---------:|---------:|---------:|---------:|------:|--------:|
| WithoutLastError | 13.24 ms | 0.262 ms | 0.493 ms | 13.03 ms |  1.00 |    0.00 |
|    WithLastError | 13.38 ms | 0.263 ms | 0.526 ms | 13.46 ms |  1.01 |    0.05 |

Reply:
@maxkatz6 , @Gillibald Do I remove SetLastError in release mode?

Reply:
All users are running a release build (unless they package debug manually) so having this check only in debug does not make much sense since it would only work for the Avalonia dev team.